pkg/cmd/server/start/start_allinone.go

```diff
@@ -209,6 +218,33 @@ func (o *AllInOneOptions) Complete() error {
 	o.NodeArgs.NodeName = strings.ToLower(o.NodeArgs.NodeName)
 	o.NodeArgs.MasterCertDir = o.MasterOptions.MasterArgs.ConfigDir.Value()
 
+	// For backward compatibility, enabling node DNS continues to start the master DNS, but the
+	// container DNS server will be the node's. However, if the user has provided an override DNSAddr,
+	// we need to honor the value if the port is not 53 and we do that by disabling node DNS.
+	if !o.IsRunFromConfig() && o.NodeArgs.Components.Enabled(ComponentDNS) {
+		dnsAddr := &o.MasterOptions.MasterArgs.DNSBindAddr
+
+		// if the user set the DNS port to anything but 53, disable node DNS
+		if dnsAddr.Provided {
+			if dnsAddr.Port == 53 {
+				// continue to run the master DNS on port 8053
+				dnsAddr.Port = 8053
```

Comment:
The change is that DNS now defaults to 53.  If you explicitly tell us 53,
we continue to use it, but set master DNS to 8053 (node will be 53).

On May 27, 2016, at 10:46 AM, Jordan Liggitt <notifications@github.com>
wrote:

In pkg/cmd/server/start/start_allinone.go
<https://github.com/openshift/origin/pull/7807#discussion_r64915713>:

> @@ -209,6 +218,33 @@ func (o *AllInOneOptions) Complete() error {
>  	o.NodeArgs.NodeName = strings.ToLower(o.NodeArgs.NodeName)
>  	o.NodeArgs.MasterCertDir = o.MasterOptions.MasterArgs.ConfigDir.Value()
>
> +	// For backward compatibility, enabling node DNS continues to start the master DNS, but the
> +	// container DNS server will be the node's. However, if the user has provided an override DNSAddr,
> +	// we need to honor the value if the port is not 53 and we do that by disabling node DNS.
> +	if !o.IsRunFromConfig() && o.NodeArgs.Components.Enabled(ComponentDNS) {
> +		dnsAddr := &o.MasterOptions.MasterArgs.DNSBindAddr
> +
> +		// if the user set the DNS port to anything but 53, disable node DNS
> +		if dnsAddr.Provided {
> +			if dnsAddr.Port == 53 {
> +				// continue to run the master DNS on port 8053
> +				dnsAddr.Port = 8053

this looks like it is changing a value explicitly set by the user. am I
misreading that?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/7807/files/74f87ae69c2d603cbe84d58b9127f31f0a5620d4#r64915713>,
or mute the thread
<https://github.com/notifications/unsubscribe/ABG_p-rr7u4iRzkJEpAZPTNY6gckEYlCks5qFwO2gaJpZM4Hp34y>
.